Dockerfile

```diff
@@ -0,0 +1,17 @@
+ARG BUILD_IMAGE
+ARG RUNTIME_IMAGE
+
+FROM $BUILD_IMAGE as builder
+
+RUN mkdir -p /build /cache /env
+COPY . /buildpack
+RUN /buildpack/bin/detect /build
+RUN /buildpack/bin/compile /build /cache /env
+
+
```

Comment:
Might want to clean up the extra blank line. (Don't hit me.)

Reply:
Actually I just realised it might be intentional, to separate the two `FROM` sections. Maybe we should add another one above the first `FROM` then. So the sections are:

1) General ARGs (shared throughout following sections)
2) `FROM $BUILD_IMAGE as builder`
3) `FROM $RUNTIME_IMAGE`

Reply:
Yeah. Agreed, I can do that or else code comments before both, though the titles are pretty much going to repeat the obvious :-)

Reply:
Cool yeah I agree comments that repeat the obvious are not good. I guess a third option would just be `#############` as a separator! At least then the comment can't go out of date :smile: But meh, not a big deal.